### ISSUE TITLE
prepare job init container resources

### ISSUE THREAD

**Issue by kfox1111** [rook/rook#6117]
* Bug Report

**Deviation from expected behavior:**
init containers on prepare jobs don't use the request/limits specified in the crd.

**Expected behavior:**
They are honored.

**How to reproduce it (minimal and precise):**
in your ceph cluster crd:
```
    prepareosd:
      limits:
        cpu: 100m
        memory: 512Mi
      requests:
        cpu: 100m
        memory: 512Mi
```
Then check on the init container in the prepare pods. they are unset.

* Rook version (use `rook version` inside of a Rook Pod):
1.3.8

**Comment by travisn** [comment]
Fixed in #6118, will be included in v1.4.2

**Comment by kfox1111** [comment]
Thank you! :)
